### ISSUE TITLE
[@angular-eslint/prefer-signals] readonly check not working for asReadonly signals

### ISSUE THREAD

**Issue by DrakeAnglin** [angular-eslint/angular-eslint#2216]
**Description and reproduction of the issue**

When enabling the prefer-signals rule, the check is ignored for any signal that is marked `asReadonly()`. This is an issue as this check is to ensure you dont overwrite your variable. The `asReadonly` function only ensures that you can not change the internal value of a signal, but the reference could still be overwritten with a new signal.

the below example does not flag during lint.
```ts
class Test {
  testSignal = signal(0).asReadonly();
}
```

This is an issue as later on in your component you can still overwrite your signal. 

```ts
class Test {
  testSignal = signal(0).asReadonly();

  foo() {
    this.testSignal = signal(1);
  }
}
```

Issues with configuration and rules should be easily reproducible on StackBlitz.

LINK HERE: https://stackblitz.com/edit/angular-eslint-repros-1tejypef?file=src%2Fmain.ts&view=editor

**Versions**

| package                         | version |
| ------------------------------- | ------- |
| `@angular-eslint/eslint-plugin` | `19.0.7` |
| `@typescript-eslint/parser`     | `7.18.0` |
| `ESLint`                        | `8.57.0` |
| `node`                          | `20.11.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 19.0.7
Node: 20.11.1
Package Manager: npm 10.2.4
OS: darwin arm64

Angular: 19.0.7
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... platform-server, router, service-worker, ssr

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1900.7
@angular-devkit/build-angular   19.0.7
@angular-devkit/core            19.0.7
@angular-devkit/schematics      19.0.7
@angular/cdk                    19.1.1
@angular/material               19.1.1
@schematics/angular             19.0.7
rxjs                            7.8.1
typescript                      5.6.3
zone.js                         0.15.0

```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by reduckted** [comment]
😕 I couldn't reproduce this in the linked StackBlitz for some reason, but it's definitely a problem. The tests I added in #2218 failed before I added the fix.
